parlai/utils/conversations.py

```diff
@@ -0,0 +1,188 @@
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Utility methods for conversations format.
+"""
+import datetime
+import json
+import os
+import random
+
+
+class Metadata:
+    """
+    Utility class for conversation metadata.
+
+    Metadata should be saved at <datapath>.metadata.
+    """
+
+    def __init__(self, datapath):
+        self._load(datapath)
+
+    def _load(self, datapath):
+        metadata_path = self._get_path(datapath)
+        if not os.path.isfile(metadata_path):
+            raise RuntimeError(
+                f'Metadata at path {metadata_path} not found. '
+                'Double check your path.'
+            )
+
+        with open(metadata_path, 'rb') as f:
+            metadata = json.load(f)
+
+        self.datetime = metadata['date']
+        self.opt = metadata['opt']
+        self.self_chat = metadata['self_chat']
+        self.speaker_1 = metadata['speaker_1']
+        self.speaker_2 = metadata['speaker_2']
+        self.extra_data = {}
+        for k, v in metadata.items():
+            if k not in ['date', 'opt', 'speaker_1', 'speaker_2']:
+                self.extra_data[k] = v
+
+    def read(self):
+        """
+        Read the relevant metadata.
+        """
+        print(f'Saved at: {self.datetime}')
+        print(f'Self chat: {self.self_chat}')
+        print(f'Speaker 1: {self.speaker_1}')
+        print(f'Speaker 2: {self.speaker_2}')
+        print('Opt:')
+        for k, v in self.opt.items():
+            print(f'\t{k}: {v}')
+        for k, v in self.extra_data.items():
+            print(f'{k}: {v}')
+
+    @staticmethod
+    def _get_path(datapath):
+        fle, _ = os.path.splittext(datapath)
+        return os.path.join(fle, '.metadata')
+
+    @classmethod
+    def save_metadata(
+        cls,
+        datapath,
+        opt,
+        self_chat=False,
+        speaker_1='human',
+        speaker_2=None,
+        **kwargs,
+    ):
+        """
+        Dump conversation metadata to file.
+        """
+        metadata = {}
+        metadata['date'] = str(datetime.datetime.now())
+        metadata['opt'] = opt
+        metadata['self_chat'] = self_chat
+        metadata['speaker_1'] = speaker_1
+        if speaker_2 is not None:
+            metadata['speaker_2'] = speaker_2
+        else:
+            metadata['speaker_2'] = opt.get('model')
+
+        for k, v in kwargs.items():
+            metadata[k] = metadata[v]
+
+        metadata_path = cls._get_path(datapath)
+        print(f'[ Writing metadata to file {metadata_path} ]')
+        with open(metadata_path, 'w') as f:
+            f.write(json.dumps(metadata))
+
+
+class Conversations:
+    """
+    Utility class for reading and writing from ParlAI Conversations format.
+
+    Conversations should be saved in JSONL format, where each line is
+    a JSON of the following form:
+    {
+        'possible_conversation_level_info': True,
```

Comment:
so what would this be? or is this just for the draft

Reply:
this is just for the draft. i left it open such that you can add whatever other information you want in there.